### ISSUE TITLE
Semantic of setObjSense is different for CPLEX and SoPlex

### ISSUE THREAD

**Issue by FlorianPommerening** [coin-or/Osi#123]
`OsiSpxSolverInterface::setObjSense` has no effect when it is called before `loadProblem` but in `OsiCpxSolverInterface`it affects the next loaded problem. I'm not sure this is the right place to report this (it might be an issue with SoPlex) but currently it means that a program can change its meaning from maximizing to minimizing when only the solver is changed. The main problem for me was that calling `setObjSense` before `loadProblem` failed silently without any indication that the sense remained unchanged.

I tried this with CPLEX 12.9, SoPlex 3.1.1, and Osi 0.107.9. if it helps, I can attach a sample program.

**Comment by svigerske** [comment]
Can you check whether the following patch fill fix this, i.e., ensure that the sense set before loadProblem survives?
```diff
--- OsiSpxSolverInterface.cpp	(revision 2269)
+++ OsiSpxSolverInterface.cpp	(working copy)
@@ -1184,6 +1184,7 @@
   const double *elem = matrix.getElements();
   const int *index = matrix.getIndices();
   double *thecollb, *thecolub, *theobj, *therowlb, *therowub;
+  soplex::SPxLP::SPxSense objsen = soplex_->spxSense();
 
   // create defaults if parameter is NULL
   if (collb == NULL) {
@@ -1266,8 +1267,8 @@
     // soplex_->changeBounds( soplex::Vector( ncols, thecollb ), soplex::Vector( ncols, thecolub ) );
   }
 
-  // switch sense to minimization problem
-  soplex_->changeSense(soplex::SPxSolver::MINIMIZE);
+  // make sure original sense is preserved
+  soplex_->changeSense(objsen);
 
   // delete default arrays if neccessary
   if (collb == NULL)
@@ -1388,6 +1389,7 @@
   int col, pos;
   soplex::LPColSet colset(numcols, start[numcols]);
   soplex::DSVector colvec;
+  soplex::SPxLP::SPxSense objsen = soplex_->spxSense();
 
   soplex_->clear();
   spxintvars_->clear();
@@ -1402,6 +1404,7 @@
 
   soplex_->addCols(colset);
   soplex_->changeRange(soplex::Vector(numrows, const_cast< double * >(rowlb)), soplex::Vector(numrows, const_cast< double * >(rowub)));
+  soplex_->changeSense(objsen);
 }
 
 //-----------------------------------------------------------------------------
```

**Comment by FlorianPommerening** [comment]
Looks fine to me, thank you. I tested this with the versions I used previously (SoPlex 3.1.1 and Osi 0.107.9), so the patch didn't apply directly but I think I found the correct lines in the old version.

Out of curiosity: what is the reason to explicitly set the sense to its old value instead of leaving it unchanged (i.e. removing all lines with ```changeSense``` in the patch)? Could it change as a side effect of some other method?
